cmd/k8s-operator/proxygroup.go

```diff
@@ -192,6 +203,32 @@ func (r *ProxyGroupReconciler) reconcilePG(ctx context.Context, pg *tsapi.ProxyG
 	return staticEndpoints, nrr, nil
 }
 
+func (r *ProxyGroupReconciler) validate(ctx context.Context, pg *tsapi.ProxyGroup) error {
+	if isAuthAPIServerProxy(pg) {
+		// Validate that the static ServiceAccount already exists.
+		sa := &corev1.ServiceAccount{}
+		if err := r.Get(ctx, types.NamespacedName{Namespace: r.tsNamespace, Name: authAPIServerProxySAName}, sa); err != nil {
+			if apierrors.IsNotFound(err) {
+				return fmt.Errorf("the ServiceAccount %q used for the API server proxy in auth mode does not exist but should have been created during operator installation", authAPIServerProxySAName)
```

Comment:
Suggestion: might be worth to log something like 'ensure that `.apiServerProxy.auth` is set to true in Helm chart values or similar, so users know how to fix this? Assuming that this invalid state could have happened as a result of user misconfiguration